### ISSUE TITLE
Process doesn't end after closing packaged app or unreal editor

### ISSUE THREAD

**Issue by khalibloo** [20tab/UnrealEnginePython#485]
Sometimes, it seems like the lingering process is stuck in an infinite loop with a steady amount of CPU being consumed indefinitely. Other times, there's 0% CPU activity from the process as only the RAM is consumed. The process still remains even after an hour from the time the application was closed.

I'm not doing any multithreading or anything fancy. I did have the QT example on it, but even after clearing all the lines in all the python scripts, just starting and closing the packaged application leaves the process behind. I have a PyActor with an empty script too (every method simply has "pass") that can be instantiated. But even without instantiating it, the process can't seem to end from just closing the app.

I haven't tried reproducing this on a new project yet. But I only began noticing this issue after packaging with UnrealEnginePython. I cannot say for sure that the issue was caused by the plugin, but it is a strong suspect. In fact, closing the unreal editor after having played in the editor leaves an unreal engine process behind as well.
![uepy issue](https://user-images.githubusercontent.com/6402969/43319003-3096d8b2-91b4-11e8-9bb2-24598264de99.png)

The processes just consume RAM quietly most of the time. But after several runs, they can pile up and have caused my machine itself to freeze, requiring a forced shutdown.

**Comment by rdeioris** [comment]
just for being sure, have you tried repackaging it without the python plugin ?

Which python version are you using ? Is this the latest release of the plugin ?

**Comment by khalibloo** [comment]
Yes, I cloned master roughly 24 hours ago. I'm using Python 3.7 (64-bit) and Unreal Engine 4.19.2. Before integrating the plugin, this issue wasn't there.

I will now try removing the plugin and re-packaging.

**Comment by khalibloo** [comment]
Ok, I can't reproduce it with just the plugin itself on a fresh blank project. Then I tried adding my QT scripts and the issue popped up. I noticed from the unreal editor logs that the QT app instance never quits between reruns in the editor. It could be the culprit after all. I tried adding `app.quit()` to my implementation but still didn't solve it. Looks like I'm doing something wrong somewhere, but curious as to why clearing the contents of my scripts in the packaged game didn't get rid of the issue.
I'll try to figure it out, but here's my ueqt script.
```
import sys
import unreal_engine as ue
import PySide2
from PySide2 import QtWidgets

class UEQT:
    app = None
    app_users_count = 0

    def ticker_loop(self, delta_time):
        self.app.processEvents()
        return True

    def add_app_user(self):
        self.app_users_count += 1
        if self.app is None:
            self.app = QtWidgets.QApplication(sys.argv)
            self.ticker = ue.add_ticker(self.ticker_loop)

    def remove_app_user(self):
        self.app_users_count -= 1
        if self.app_users_count == 0:
            self.app.quit()
            ue.log("QT application closed")

inst = UEQT()
```
and my file_dialog script
```
from ueqt import inst as ueqt
from PySide2 import QtCore, QtWidgets, QtGui
import unreal_engine as ue
from PySide2.QtWidgets import QFileDialog
from unreal_engine import FSlateApplication


class DummyWidget(QtWidgets.QWidget):
    def __init__(self):
        super().__init__()

class FileDialogManager:

    # this is called on game start
    def begin_play(self):
        ueqt.add_app_user()
        
    def open(self):
        widget = DummyWidget()
        # widget.resize(800, 600)
        # widget.show()

        # get the active top level window
        top_window = FSlateApplication.get_active_top_level_window()
        top_window.set_as_owner(widget.winId())

        f = QFileDialog.getOpenFileName(
            widget, #parent
            self.uobject.Title, #title
            self.uobject.InitialDir, #initial dir
            ";;".join(self.uobject.FileTypeFilters), #filters
            self.uobject.FileTypeFilters[self.uobject.SelectedFileTypeFilter] #selected filter
        )
        if len(f[0]) < 1:
            ue.log("Open File Dialog:::Canceled")
            self.uobject.call_function('OnCancel')
            # sys.exit()
            return
        ue.log("Open File Dialog:::Confirmed")
        ue.log(f)
        widget.close()
        self.uobject.Filepath = f[0]
        self.uobject.call_function('OnOpen')
        # sys.exit()
        
    # this is called at every 'tick'
    def tick(self, delta_time):
        pass
    
    # this is called at the end
    def end_play(self, reason=None):
        ue.log("File Dialog:::Quit")
        ue.log(reason)
        ueqt.remove_app_user()
```
The PyActor simply calls "open" on begin play.

**Comment by rdeioris** [comment]
Hi, can you try with the latest release ? A bunch of fixes have been added after a very similar report (albeit not related to qt)

**Comment by khalibloo** [comment]
Awesome! Will report back hopefully tonight.

**Comment by khalibloo** [comment]
I pulled from master and rebuilt in visual studio. The issue is still present.

**Comment by rdeioris** [comment]
So, after a bit more digging i found that the problem is PySide itself. Even if you do not call any of its api functions, just importing its modules will break the Py_Finalize() procedure.

For this reason i have added ue.set_brutal_finalize(bool) that will skip the call to Py_Finalize():

https://github.com/20tab/UnrealEnginePython/commit/8308e441c10fd057e348dc830a367c01da9153b6
